### DISCUSSION TITLE
apache-answer and discourse.org

### DISCUSSION THREAD

**Discussion by rajesh99** [Southclaws/storyden#175] [Q&A]
Have you looked at apache-answer and discourse.org
How will storyden be different?

**Comment by Southclaws** [comment]
Hey @rajesh99!

Yes I've looked into both, and I've hosted Discourse before for a rather large community.

They both serve some similar and some different objectives to Storyden:

- Apache Answer is primarily a Stack Exchange style Q&A platform, it's primarily built for this objective in mind (based on looking at the code) and seems to provide a fairly nice experience for that goal. It isn't really a discussion forum or a knowledge-base and it seems they've focused just on the Q&A product at this time.
- Discourse is the go-to solution for support forums, it has Q&A features too and their market focus is primarily enterprise and is often the choice for larger companies who want to set up a forum for their customers to ask support questions to other users. I've also seen it used as a bulletin board but this is less common.
  - I've hosted Discourse before and it's very "heavy" and not necessarily a simple piece of software to deploy in modern infrastructure (orchestrated Docker containers or onto services like Fly.io) though this was a few years ago so it could have improved since. Frustration with Discourse was actually one of the original reasons why our community built their own forum which then evolved into a standalone product: Storyden! Discourse is a great product, but I think their focus lies less on the open source product and more on the paid hosted service.

Storyden differs in a few ways:
- Minimal by default: You can set up a production grade instance by deploying a single binary with no external services required (SQLite database, filesystem storage)
- Scales if you need it: You can use Postgres, S3 for storage and AMQP for background jobs if the need arises and scale horizontally for high traffic instances.
- Cloud native: Built for modern infrastructure if you want, while you can drop a single binary on a VPS and get a forum up in a few seconds, you can also deploy to a cattle-oriented cloud provider (Fly, K8s, Docker, etc) with all the expected features: stateless instance, environment variable driven configuration, metrics, etc.
- AI-first: it has (optional) LLM features built directly in so operators and members can harness useful AI features such as search, questions and recommendations 
- Flexible and generic: The backend is essentially a CMS for social products with a powerful API and features that can facilitate a number of use-cases from building your own Reddit to building your own social e-commerce site. The frontend provided in the `./web` directory is merely a reference for one use-case. Storyden's backend is currently being used in two other products that aren't traditional "forum" products but benefit greatly from the social and LLM features. You can think of it as a WordPress for the modern age.

Currently, it's just me building with some help from friends architecting and supporting so there are some big plans as we grow! I hope that answers your questions, thanks!

**Comment by rajesh99** [comment]
Wonderful!

I am exploring and finding the differences and similarities of some software.  

I have asked similar questions on apache-answer also.

I am looking at doing some integrations with 
- my field service saas
- integrate with google workspace and offer on their google workspace market place 

I have mostly worked on GCP products like datastore and firestore.    I might see on how to use firestore as a db.
